src/mame/sinclair/sprinter.cpp

```diff
@@ -1506,10 +1521,20 @@ void sprinter_state::do_cpu_wait(bool is_io)
 
 TIMER_CALLBACK_MEMBER(sprinter_state::irq_on)
 {
-	m_maincpu->pulse_input_line(INPUT_LINE_IRQ0, attotime::from_ticks(26, m_maincpu->clock()));
+	if (!m_hold_irq)
+	{
+		m_maincpu->set_input_line(INPUT_LINE_IRQ0, ASSERT_LINE);
+		m_irq_off_timer->adjust(attotime::from_ticks(32, m_maincpu->unscaled_clock()));
+	}
 	update_int(false);
 }
```

Comment:
Should this interrupt continue to be held for the full 32 cycles even if there’s an interrupt acknowledge cycle, or should an interrupt acknowledge cycle clear this interrupt as well as the other one?  (The implementation currently does the latter – clears both kinds of interrupt on an interrupt acknowledge cycle.)